discovery-infra/log_scrap.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+import re
+import os
+import time
+import json
+import yaml
+import random
+import urllib3
+import logging
+import hashlib
+import tempfile
+import elasticsearch
+
+from logger import log
+from shutil import copyfile
+from datetime import datetime
+from filelock import FileLock
+from monitoring import process
+from contextlib import suppress
+from argparse import ArgumentParser
+from contextlib import contextmanager
+
+import assisted_service_client
+from test_infra.assisted_service_api import InventoryClient, create_client
+
+RETRY_INTERVAL = 60 * 5
+MAX_EVENTS = 1000
+INDEX = "assisted-service-events"
+
+FMT = '%Y-%m-%dT%H:%M:%S.%fZ'
+UUID_REGEX = r'[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}'
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+es_logger = logging.getLogger('elasticsearch')
+es_logger.setLevel(logging.WARNING)
+
+class ScrapeEvents:
+    def __init__(self, inventory_url: str, es_server: str, es_user:str, es_pass:str, backup_destination: str):
+        self.client = create_client(url=inventory_url)
+
+        self.index = INDEX
+        self.es = elasticsearch.Elasticsearch(es_server, http_auth=(es_user, es_pass))
+
+        self.backup_destination = backup_destination
+        if self.backup_destination and not os.path.exists(self.backup_destination):
+            os.makedirs(self.backup_destination)
+
+
+    def run_service(self):
+
+        while True:
+            clusters = self.get_clusters()
+            random.shuffle(clusters)
+
+            if not clusters:
+                log.warn(f'No clusters were found, waiting {RETRY_INTERVAL/60} min')
+                time.sleep(RETRY_INTERVAL)
+                break
+
+            cluster_count = len(clusters)
+            for i, cluster in enumerate(clusters):
+                cluster_id = cluster["id"]
+                log.info(f"{i}/{cluster_count}: Starting process of cluster {cluster_id}")
+                self.process_cluster(cluster)
+
+    def get_last_cluster_event_time(self, cluster_id: str):
+        try:
+            res = self.get_last_cluster_event(cluster_id)
+        except elasticsearch.exceptions.NotFoundError:
+            return None
+        if not res["hits"]["hits"]:
+            return None
+        return res["hits"]["hits"][0]["_source"]["event_time"]
+
+    def get_last_cluster_event(self, cluster_id):
+        res = self.es.search(index=self.index, body={
+            "size": 1,
+            "sort": {"event_time": "desc"},
```

Comment:
For cases where you only want the `event_time` you can pass `_source: ['event_time']` which will only return that field.